src/general/cpuInfo.go

```diff
@@ -47,30 +48,53 @@ func NewCPULoad() *CPULoad {
 	return &CPULoad{}
 }
 
+// ReadCPULoad reads /proc/stat and returns the average load
+func (c *CPULoad) readCPULoad() error {
+	file, err := os.Open("/proc/stat")
+	if err != nil {
+		return err
+	}
+	defer file.Close()
```

Comment:
```suggestion
	file, err := os.Open("/proc/stat")
	defer file.Close()
	if err != nil {
		return err
	}
```

Reply:
Yeah, it did warn me that I should probably check for the error before calling the file.Close() function, which I why I have the if conditions first. 

Yep. I shall add comments tomorrow!

Reply:
Warned you for having `defer` before the `if`?

Reply:
No the warning only highlighted 'file.close()'. It wasn't a compiler warning, it was the linter

Reply:
@MadhavJivrajani you only want the defer to be called when `err != nil`, else when error is not nil and we return the error, the close function will be called. This could cause a panic as you're trying to close a file which errored when opening. So I'd avoid changing this. It's good as is.

Reply:
Ah I see, now that I think about it, the way you've done it is fine, if there is an error, file wouldn't be open anyway and the Go blog also uses it in a similar way you have

Reply:
> 

@MadhavJivrajani you only want the defer to be called when err != nil, else when error is not nil and we return the error, the close function will be called. This could cause a panic as you're trying to close a file which errored when opening. So I'd avoid changing this. It's good as is.

Makes sense.